ioos_catalog/views/resolver.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+'''
+ioos_catalog/views/resolver.py
+'''
+
+from ioos_catalog import app, db
+from bson import json_util
+from flask import jsonify, render_template, url_for
+from glob import glob
+from ioos_catalog.views.ra import provider_info as p_info
+from collections import OrderedDict
+import json
+
+def dot_get(o, path):
+    i = path.split('.', 1)
+    if len(i) > 1:
+        return dot_get(o[i[0]], i[1])
+    return o[i[0]]
+
+def load_assets(key, path):
+    assets_json = 'Assets.json'
+    with open(assets_json, 'r') as f:
+        assets = json.loads(f.read())
+
+    assets = dot_get(assets, key)
+    files = []
+    for filepath in assets[path]:
+        if '*' not in filepath:
+            files.append(filepath)
+        else:
+            for glob_path in glob(filepath):
+                files.append(glob_path)
+    # Strip off the beginning piece
+    files = [f.replace('ioos_catalog', '') for f in files]
+    return files
+
+def load_javascripts(key, template_name):
+    path = 'ioos_catalog/static/js/compiled/%s.js' % template_name
+    if app.config['DEBUG']:
+        scripts_list = []
+        for js_file in load_assets(key, path):
+            script = '<script src="%s" type="text/javascript"></script>' % js_file
+            scripts_list.append(script)
+        scripts = '\n'.join(scripts_list)
+        return scripts
+    path = path.replace('ioos_catalog', '')
+    return '<script src="%s" type="text/javascript"></script>' % path
+
+def load_css(key, template_name):
+    path = 'ioos_catalog/static/css/compiled/%s.css' % template_name
+    if app.config['DEBUG']:
+        scripts_list = []
+        for css_file in load_assets(key, path):
+            script = '<link href="%s" rel="stylesheet" type="text/css" />' % css_file
+            scripts_list.append(script)
+        scripts = '\n'.join(scripts_list)
+        return scripts
+    path = path.replace('ioos_catalog', '')
+    return '<link href="%s" rel="stylesheet" type="text/css" />' % path
+
+@app.route('/api/resolver/config.json', methods=['GET'])
+def get_resolver_config():
+    navbar_img = url_for('.static', filename='img/ioos.png')
+    inventory_link = url_for('inventory')
+    services_link = url_for('services')
+    datasets_link = url_for('datasets')
+    gliders_link = url_for('gliders')
+    metadatas_link = url_for('metadatas')
+    catalog_map_link = url_for('catalog_map')
+    help_link = url_for('help')
+
+    return jsonify(locals())
+
+@app.route('/api/providers', methods=['GET'])
+@app.route('/api/providers/', methods=['GET'])
+def get_providers():
```

Comment:
Starting to add a RESTful interface (unlikely to be full-rest in the future, I doubt we'll support anything but GET, but who knows...)